### PR TITLE
Replace librarian-chef with berkshelf

### PR DESCRIPTION
**Pull request by cwjohnston**
## Description

Replace librarian-chef with berkshelf for this project's cookbook dependency resolution.

## Motivation and Context

librarian-chef currently fails to resolve our dependencies. This
seems to have something to do with its graphs for erlang and various yum*
cookbook dependencies but I haven't been able to zero in on a satisfactory
cause.

## How Has This Been Tested?

Unit tests pass, subset of test-kitchen runs pass.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ group :develop do
   gem "chef", "~> 12.9"
   gem "chefspec", "~> 4.5"
   gem "emeril", "~> 0.8"
-  gem "librarian-chef"
+  gem "berkshelf", "~> 5.0"
```
Comment (tas50) [inline Gemfile]: Be aware this won't install within the current ChefDK since it's Ruby 2.1.9 and this requires Ruby 2.2+

Comment (cwjohnston) [inline Gemfile]: What version dependency do you recommend?

Comment (cwjohnston) [inline Gemfile]: @tas50 would < 6.0 make sense?

**Comment by tas50** [comment]
>= 4.3.0 <6 is probably a good choice.

**Comment by cwjohnston** [comment]
I'll point out that Batali also works for resolving our dependencies, but doesn't enjoy first-class support in Test Kitchen 😿

**Comment by portertech** [comment]
Looks good, pending the the berks version constraint change :+1:

**Pull request merged**